README.rst

```diff
@@ -1,10 +1,161 @@
+^^^^^^^^^^^^^^^^^^^
+
+If you are testing machines that use ``https`` but do not have good
+certificates, it can get difficult to read the test results due to warnings
+about insecure connections. If this is a problem for you, an example
+``pytest.ini`` file is provided in the root of the ``Camayoc`` repository that
+disables these warnings. Additional configuration is possible through the
+``pytest.ini`` file, these options are documented at
+https://docs.pytest.org/en/latest/reference.html#ini-options-ref
+
+For example, if you always want to run ``py.tests`` in verbose mode, this can
+be done in the ``pytest.ini`` file.
+
+Running The Test Suite
+^^^^^^^^^^^^^^^^^^^^^^
+
+While Camayoc  contains test for rho_ and quipucords_, and these share
+a config file format, testing both rho_ and quipucords_ simultaneously is
+not a supported use case.
+
+Camayoc uses ``py.test`` as its test driver. The pytest project had excellent
+documentation available at https://docs.pytest.org/en/latest/contents.html
+
+Running Tests For rho_
+^^^^^^^^^^^^^^^^^^^^^^
+
+To run all the tests for rho_, first have rho_ installed and available in
+your ``$PATH``. This can be done by either ``rpm`` install to the system or by
+``pip`` installing rho_ from source into your Camayoc virtual
+environment. Then invoke the main test suite from the root directory of
+Camayoc with::
+
+    make test-rho
+
+There are some additional tests for rho located in ``camayoc/tests/remote/rho``
+that scan all machines listed in the inventory with all available network
+credentials. To run these, you similarly invoke::
+
+    make test-rho-remote
+
+Running Tests For quipucords_ and qpc_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To run all the quipucords_ tests, first start the quipucords_ server and
+set up your config file to point to the server. You should also install qpc_
+in such a way that it is available in your ``$PATH``. Then run all API, CLI,
+and UI tests with::
+
+    make test-qpc
+
+.. warning::
+    There are extra considerations you must take if you are running the
+    quipucords server in a container.  First, your config file must use
+    paths for sshkeys that exist on the server.
+    Additionally you must map your ``/tmp`` directory to the server's ``tmp``
+    directory  because some tests create temporary files that must be accessed
+    by the server.
+
+To only test the API, CLI, or UI, tell pytest this by extending the path to
+only point to one of these subdirectories::
+
+    # for API tests only
+    make test-qpc-api
+    # for CLI tests only
+    make test-qpc-cli
+    # for UI tests only
+    make test-qpc-ui
+
+The tests can be further selected by specifying any path or file. Additionally
+you can select tests based on string matching. For example, to run quipucords
+tests with ``create`` in the name and skip any others, run::
+
+    make test-qpc-filter pattern=create
+
+Or to select a specific test, type out the whole name of the test.
```

Comment:
I think worth adding a note here, specifying the hole name does not ensure a single test to be run, if `test_something` exists in more than one file all tests will be run.